sentry-apache-http-client-5/src/main/java/io/sentry/transport/apache/ApacheHttpClientTransport.java

```diff
@@ -71,68 +85,105 @@ public void send(final @NotNull SentryEnvelope envelope, final @Nullable Map<Str
       final SentryEnvelope filteredEnvelope = rateLimiter.filter(envelope, sentrySdkHint);
 
       if (filteredEnvelope != null) {
-        currentlyRunning.increment();
-
-        try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
-          options.getSerializer().serialize(filteredEnvelope, gzip);
-
-          final SimpleHttpRequest request =
-              SimpleHttpRequests.post(requestDetails.getUrl().toString());
-          request.setBody(
-              outputStream.toByteArray(), ContentType.create("application/x-sentry-envelope"));
-          request.setHeader("Content-Encoding", "gzip");
-          request.setHeader("Accept", "application/json");
-
-          for (Map.Entry<String, String> header : requestDetails.getHeaders().entrySet()) {
-            request.setHeader(header.getKey(), header.getValue());
-          }
+        final SentryEnvelope envelopeWithClientReport =
```

Comment:
`l`: Not related to this PR, but the code contains deeply nested if conditions and it's hard to read for me. Maybe we could use a few guard clauses / return statements to reduce the nesting. If we do, please not in this PR. @marandaneto

Reply:
@adinauer Maybe check with @maciejwalkowiak if it's worth refactoring this class